addons/website_sale/controllers/product_configurator.py

```diff
@@ -1,147 +1,263 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
+
 import json
 
 from odoo.http import request, route
+from odoo.tools import float_is_zero
+
+from odoo.addons.sale.controllers.product_configurator import SaleProductConfiguratorController
 from odoo.addons.website_sale.controllers.main import WebsiteSale
 
-class WebsiteSaleProductConfiguratorController(WebsiteSale):
+
+class WebsiteSaleProductConfiguratorController(SaleProductConfiguratorController, WebsiteSale):
 
     @route(
-        '/sale_product_configurator/show_advanced_configurator',
-        type='json', auth='public', methods=['POST'], website=True,
+        route='/website_sale_product_configurator/show_dialog',
+        type='json',
+        auth='public',
+        website=True,
     )
-    def show_advanced_configurator(
-        self, product_id, variant_values, add_qty=1, force_dialog=False, **kw,
+    def website_sale_product_configurator_show_dialog(
+        self, product_template_id, ptav_ids, is_product_configured
     ):
-        product = request.env['product.product'].browse(int(product_id))
-        product_template = product.product_tmpl_id
-        combination = request.env['product.template.attribute.value'].browse(variant_values)
-        has_optional_products = product.optional_product_ids.filtered(
-            lambda p: p._is_add_to_cart_possible(combination)
-                      and (not request.website.prevent_zero_price_sale or p._get_contextual_price())
-        )
+        """ Return whether the product configurator dialog should be shown.
 
-        already_configured = bool(combination)
-        if not force_dialog and not has_optional_products and (
-            product.product_variant_count <= 1 or already_configured
-        ):
-            # The modal is not shown if there are no optional products and
-            # the main product either has no variants or is already configured
-            return False
-
-        add_qty = float(add_qty)
-        combination_info = product_template._get_combination_info(
-            combination=combination,
-            product_id=product.id,
-            add_qty=add_qty,
+        :param int product_template_id: The product being checked, as a `product.template` id.
+        :param list(int) ptav_ids: The combination of the product, as a list of
+                                   `product.template.attribute.value` ids.
```

Comment:
I'm fine with your suggestion if that's what we do elsewhere, but I noticed we also use `combination` for PTAV recordsets, which might be a bit confusing (because we reassign the variable from a list of ids to a recordset).

BTW, `sale_product_configurator_get_values` also uses `ptav_ids`, so I'll standardize that one as well.

Reply:
Ok I did not see it that way, maybe `ptav_ids` is better then update the docstrings to be clearer since it's not the combination of the product but the list of selected attribute ids